src/platform/user/profile/vet360/containers/AddressValidationModal.jsx

```diff
@@ -0,0 +1,177 @@
+
+    return addressValidationError
+      ? 'We’re sorry. It looks like the address you entered isn’t valid.  Please edit your address'
+      : 'We’re sorry. It looks like the address you entered isn’t valid.  Please enter your address again or choose a suggested address below';
+  };
+
+  renderPrimaryButton = () => {
+    const {
+      addressValidationError,
+      addressValidationType,
+      validationKey,
+    } = this.props;
+
+    if (addressValidationError && !!validationKey) {
+      return (
+        <button
+          className="usa-button-primary"
+          onClick={() => this.props.openModal(addressValidationType)}
+        >
+          Edit Address
+        </button>
+      );
+    }
+
+    return <button className="usa-button-primary">Continue</button>;
+  };
+
+  renderAddressOption = (address, id = 'userEntered') => {
+    const { validationKey } = this.props;
+    const {
+      addressLine1,
+      addressLine2,
+      addressLine3,
+      city,
+      stateCode,
+      zipCode,
+    } = address;
+
+    const isOptionDisabled = id === 'userEntered' && !!validationKey;
+
+    return (
+      <div key={id}>
+        <input
+          style={{ zIndex: '1' }}
+          type="radio"
+          name={id}
+          disabled={isOptionDisabled}
+          checked={this.state.selectedOption === id}
+          onClick={this.onChangeHandler(address)}
+        />
+        <label
+          htmlFor={id}
+          style={{
+            marginTop: '7px',
+            display: 'flex',
+            alignItems: 'center',
+          }}
+        >
+          <div style={{ display: 'flex', flexDirection: 'column' }}>
+            {addressLine1 && <span>{addressLine1}</span>}
+            {addressLine2 && <span>{addressLine2}</span>}
+            {addressLine3 && <span>{addressLine3}</span>}
+            <span>{`${city}, ${stateCode} ${zipCode}`}</span>
+          </div>
+        </label>
+      </div>
+    );
+  };
+
+  render() {
+    const {
+      closeModal,
+      isAddressValidationModalVisible,
+      addressValidationType,
+      addresses,
+    } = this.props;
+
+    const addressFromUser =
+      this.props.userEnteredAddress[`${addressValidationType}`]?.value || {};
+
+    return (
+      <Modal
+        title={
+          addressValidationType.includes('mailing')
+            ? 'Edit mailing address'
+            : 'Edit home address'
+        }
+        id="address-validation-warning"
+        onClose={closeModal}
+        visible={isAddressValidationModalVisible}
+      >
+        <AlertBox status="warning" headline="Your address update isn't valid">
+          <p>{this.renderWarningText()}</p>
+        </AlertBox>
+        <form onSubmit={this.onSubmit}>
+          <span className="vads-u-font-weight--bold">You entered:</span>
+          {this.renderAddressOption(addressFromUser)}
+          <span className="vads-u-font-weight--bold">Suggested Addresses:</span>
+          {map(addresses, (address, index) =>
```

Comment:
if we used native `map` and we're concerned about `addresses` being undefined, we could just do something like:

```js
addresses?.map((address, index) => this.renderAddressOption(address, String(index)))
```

And if we're worried about `addresses` being undefined we probably want to check for that before rendering `Suggested Addresses:` on line 139 above.